renderdoc/driver/metal/metal_init_state.cpp

```diff
@@ -39,16 +102,44 @@ bool WrappedMTLDevice::Prepare_InitialState(WrappedMTLObject *res)
 
 uint64_t WrappedMTLDevice::GetSize_InitialState(ResourceId id, const MetalInitialContents &initial)
 {
-  METAL_NOT_IMPLEMENTED();
-  return 128;
+  uint64_t ret = 12;
```

Comment:
I wasn't sure if it was helpful to increase the size to allow for future changes or tight fitting.
Currently, this size is tight fitting.

Reply:
The reason for the size estimate is so that chunks written directly through a compressor to disk can have their size given without needing to seek/rewind in the stream to patch the size afterwards to the tight size, and without needing to serialise temporarily in-memory until the length is known then copy.

As long as the size estimate is greater or equal to the actual size, everything will work you'll just get some amount of padding (which should compress well as it's initialised to 0), so realistically for chunks that are often MBs in size having a few hundred bytes of padding is not any issue and it's better to be sure that your estimate is large enough than have a strict tight bound. Especially true for more complex serialisation where the overhead of internal array sizes before arrays/buffers is less clear.